<a href="https://colab.research.google.com/github/benteMarij/DSRP-Small-Body-Close-Approaches-BMdZ/blob/main/Bente_de_Zwart_DSRP_Colab_Small_Body_Close_Approaches.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 50)

In [22]:
df = pd.read_csv('https://raw.githubusercontent.com/frontiertechinstitute/datasets/refs/heads/main/SB%20Close%20Approaches/earth_close_approaches_50yr.csv')
df.info()
df

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 186255 entries, 0 to 186254
Data columns (total 8 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   des       186255 non-null  object 
 1   cd        186255 non-null  object 
 2   dist      186255 non-null  float64
 3   dist_min  186255 non-null  float64
 4   dist_max  186255 non-null  float64
 5   v_rel     186255 non-null  float64
 6   h         186065 non-null  float64
 7   diameter  5375 non-null    float64
dtypes: float64(6), object(2)
memory usage: 11.4+ MB


,des,cd,dist,dist_min,dist_max,v_rel,h,diameter
0,2015 RT1,1976-Jan-01 04:35,0.111611,0.109327,0.113897,11.086209,25.410,NaN
1,2025 JH2,1976-Jan-01 06:22,0.171429,0.171313,0.171548,5.842427,27.050,NaN
2,2001 KU66,1976-Jan-01 14:29,0.440401,0.426042,0.466510,10.401878,24.100,NaN
3,2015 TG24,1976-Jan-01 15:04,0.451030,0.425241,0.477393,12.519296,26.680,NaN
4,2002 TX59,1976-Jan-02 01:26,0.459706,0.459668,0.459744,6.299898,23.900,NaN
...,...,...,...,...,...,...,...,...
186250,2026 LD1,2026-Jun-12 19:50,0.130207,0.128030,0.132384,10.184379,23.314,NaN
186251,2026 LK1,2026-Jun-12 21:43,0.005988,0.005983,0.005993,5.837560,27.872,NaN
186252,388945,2026-Jun-12 22:12,0.112264,0.112264,0.112264,7.484532,20.440,NaN
186253,2018 NC15,2026-Jun-12 22:50,0.131353,0.131353,0.131354,10.225792,21.810,NaN


**Possible Issues**

May be some issues with the 'des' column, there are 2 formats for the name of the small body object. Can be either their provisional or permanent name.
* Ex: line 0 - 2015 RT1 (provisional), line 15 - 523808 (permanent)

May be some issues with the 'diameter' column, as many of the values are NaN.
* 5375 non-null out of 186255 lines (dtype: float64)
